### ISSUE TITLE
how to get the mAP graph?

### ISSUE THREAD

**Issue by shawntyshawny** [tensorflow/models#9208]
Have i written custom code?
 -No
OS Platform and Distribution:
 -Windows10
TensorFlow installed from:
 -pip install tensorflow
TensorFlow Version:
 -1.15.0
Bazel Version:
 -N/A
CUDA/cuDNN version:
 -10.0/7.6.5
GPU model and memory:
 NVIDIA GTX 2060s
Exact command to reproduce:
 python train.py --logtostderr --train_dir=training/ --pipeline_config_path=training/faster_rcnn_inception_v2_pets.config

Question:
train.py provides the losses like total loss, regional loss and classification loss, does running the model_main.py instead of train.py provides the mAP graph? if not how to get the mAP graph during training?

**Comment by ravikyram** [comment]
@shawntyshawny 

Please, go through the [link](https://towardsdatascience.com/detailed-tutorial-build-your-custom-real-time-object-detector-5ade1017fd2d) and see if it helps you. Thanks!

**Comment by shawntyshawny** [comment]
> @shawntyshawny
> 
> Please, go through the [link](https://towardsdatascience.com/detailed-tutorial-build-your-custom-real-time-object-detector-5ade1017fd2d) and see if it helps you. Thanks!


> @shawntyshawny
> 
> Please, go through the [link](https://towardsdatascience.com/detailed-tutorial-build-your-custom-real-time-object-detector-5ade1017fd2d) and see if it helps you. Thanks!

Thank you for the response, Sir! About to try this within 2 days after I fix some monitor issues. I'll update as soon as I can. Thank you sir!

**Comment by shawntyshawny** [comment]
Managed to make mAP graph show using model_main.py 

1. How can I show this training and validation loss in a single graph

![image](https://user-images.githubusercontent.com/70835409/92925836-b4ca7080-f46d-11ea-9abf-cb5cba1d300e.png)

This is what I see in my tensorboard 

![image](https://user-images.githubusercontent.com/70835409/92925881-c875d700-f46d-11ea-818e-8d3b5c161241.png)

2. My loss_1 and loss_2 in the above training progress is spiking where it should look smooth like this. My loss graph using train.py is similar to the images below.

![image](https://user-images.githubusercontent.com/70835409/92926042-070b9180-f46e-11ea-9729-1728ee86bc01.png)
![image](https://user-images.githubusercontent.com/70835409/92926118-20144280-f46e-11ea-9f42-c0bdae460ce1.png)


3. My mAP graph is so low @ 29 mAP. Others training has 80 above mAP. I'm using faster rcnn inception v2 coco
![image](https://user-images.githubusercontent.com/70835409/92926154-2f938b80-f46e-11ea-8980-a142008286ef.png)

**Comment by KevinCS17** [comment]
sorry but I still dont get it how to add mAP, can you show me more specific to add that mAP?

**Comment by shawntyshawny** [comment]
@KevinCS17 
> sorry but I still dont get it how to add mAP, can you show me more specific to add that mAP?

I just ran the model_main.py script for training. 
`python model_main.py --model_dir=training --pipeline_config_path=training/faster_rcnn_inception_v2_pets.config`
take note that model_main.py uses --model_dir=training. Also mAP graph will show after the first eval (10mins I think)

**Comment by KevinCS17** [comment]
but if I using tensorflow 2, is it okay?
model_main.py using tensorflow v1 right?
ooohh Youre tensorflow is V1..

how about V2?
is it okay?

**Comment by shawntyshawny** [comment]
> but if I using tensorflow 2, is it okay?
> model_main.py using tensorflow v1 right?
> ooohh Youre tensorflow is V1..
> 
> how about V2?
> is it okay?

Haven't tried version 2 yet.

**Comment by ravikyram** [comment]
@shawntyshawny 

Any update on this issue please. Thanks!

**Comment by sekartanjung** [comment]
Hello everyone, I want to ask about the graph of the result after train using TensorfloBoard.  The graph shows 2 lines, they are transparent line and a bold line. What is the meaning of both lines?
